apps/snitch_api/lib/snitch_api_web/controllers/product_controller.ex

```diff
@@ -0,0 +1,106 @@
+defmodule SnitchApiWeb.ProductController do
+  use SnitchApiWeb, :controller
+
+  alias Snitch.Repo
+  alias Snitch.Data.Schema.Product
+
+  import Ecto.Query, only: [from: 2]
+
+  # sort the products in Z-A descending order_by prodcut name
+  def index(conn, %{"sort" => "Z-A"} = params) do
+    query = from(p in Product, order_by: [desc: p.name])
```

Comment:
Why no use of domain functions. Why import Ecto here ?

Reply:
For developing non destructive queries

Reply:
@blackode How about we create some context to wrap these functions can call them from here.

Reply:
@blackode use context, fallback controller that phoenix 1.3 + provides.